apps/worker-service/src/app/workflow/services/workflow-queue.service.spec.ts

```diff
@@ -1,69 +1,112 @@
 import { Test } from '@nestjs/testing';
+import { expect } from 'chai';
+import { formatISO } from 'date-fns';
+import { v4 as uuid } from 'uuid';
+import { faker } from '@faker-js/faker';
+
 import {
+  EnvironmentEntity,
   JobEntity,
   JobRepository,
   JobStatusEnum,
   MessageTemplateEntity,
   NotificationRepository,
   NotificationTemplateEntity,
   NotificationTemplateRepository,
+  OrganizationEntity,
   SubscriberEntity,
+  UserEntity,
 } from '@novu/dal';
 import { StepTypeEnum } from '@novu/shared';
-import { SubscribersService, UserSession } from '@novu/testing';
-import { expect } from 'chai';
-import { formatISO } from 'date-fns';
-import { v4 as uuid } from 'uuid';
-
-import { WorkflowQueueService } from './workflow-queue.service';
+import {
+  EnvironmentService,
+  NotificationTemplateService,
+  OrganizationService,
+  SubscribersService,
+  UserService,
+  JobsService,
+} from '@novu/testing';
+import { QueueService } from '@novu/application-generic';
 
-import { WorkflowModule } from '../../workflow/workflow.module';
+import { WorkflowModule } from '../workflow.module';
 
-let workflowQueueService: WorkflowQueueService;
+let queueService: QueueService;
 
 describe('Workflow Queue service', () => {
   let jobRepository: JobRepository;
-  let session: UserSession;
+  let organization: OrganizationEntity;
+  let environment: EnvironmentEntity;
+  let user: UserEntity;
   let subscriber: SubscriberEntity;
   let subscriberService: SubscribersService;
   let template: NotificationTemplateEntity;
+  let jobsService: JobsService;
 
-  beforeEach(async () => {
-    const moduleRef = await Test.createTestingModule({
-      imports: [WorkflowModule],
-    }).compile();
+  before(async () => {
+    jobRepository = new JobRepository();
 
-    workflowQueueService = moduleRef.get<WorkflowQueueService>(WorkflowQueueService);
+    jobsService = new JobsService();
 
-    jobRepository = new JobRepository();
-    session = new UserSession();
-    await session.initialize();
+    const userService = new UserService();
+    const card = {
+      firstName: faker.name.firstName(),
+      lastName: faker.name.lastName(),
+    };
+    const userEntity: Partial<UserEntity> = {
+      lastName: card.lastName,
+      firstName: card.firstName,
+      email: `${card.firstName}_${card.lastName}_${faker.datatype.uuid()}@gmail.com`.toLowerCase(),
+      profilePicture: `https://randomuser.me/api/portraits/men/${Math.floor(Math.random() * 60) + 1}.jpg`,
+      tokens: [],
+      password: '123Qwe!@#',
+      showOnBoarding: true,
+    };
+
+    user = await userService.createUser(userEntity);
```

Comment:
I refactored this tests because before we've used here the `UserSession` which calls the API to get token and establish session, but it's not needed for this test